test/tests.dart

```diff
@@ -55,6 +55,14 @@ main() {
       return pool.getConnection().then((conn) {
         expect(conn.conn.id, equals(size));
       });
+    }).then((_) {
```

Comment:
I used the same coding style as the legacy code, as I don't have time to rewrite all.